src/extension/index.spec.ts

```diff
@@ -10,13 +10,17 @@ import { postSelectLog } from '../panel/indexStore';
 import { log } from '../test/mockLog';
 import { mockVscode, mockVscodeTestFacing, uriForRealFile } from '../test/mockVscode';
 import { URI as Uri } from 'vscode-uri';
+import { Api } from './index.d';
 
 // Log object may be modified during testing, thus we need to keep a clean string copy.
 const mockLogString = JSON.stringify(log, null, 2);
 
 const proxyquire = require('proxyquire').noCallThru();
 
-describe('activate', () => {
+let api: Api;
+
+// TODO Tests are hanging on CI.
```

Comment:
How could you tell this suite is the one that's hanging from the logs? Nothing really stood out to me.

Reply:
It was an educated guess since this was the suite that was hanging locally for me until I added the correct `dispose()` method and called it from the tests.

I'm guessing that there is something that is not being disposed properly on linux. Possibly, this is something about file watchers (since this is os-dependent and it is passing for me on MacOS). Figuring this out will take a lot of trial and error and be very time consuming.

I'll update the comment with some of my thoughts on what might be happening so it will be easier to pick up in the future.